src/pykeen/losses.py

```diff
@@ -510,6 +512,144 @@ def forward(
+
+    Despite its similarity to the margin-based loss, this loss is quite different to it, since it uses absolute margins
+    for positive/negative scores, rather than comparing the difference. Hence, it has a natural decision boundary
+    (somewhere between the positive and negative margin), while still resulting in sparse losses with no gradients for
+    sufficiently correct examples.
+
+    # TODO: Proper citation
+
+    Bootstrapping Entity Alignment with Knowledge Graph Embedding
+    Zequn Sun, Wei Hu∗, Qingheng Zhang and Yuzhong Qu
+    Proceedings of the Twenty-Seventh International Joint Conference on Artificial Intelligence (IJCAI-18)
+
+
+    .. math ::
+        L(score^+, score^-) = activation(margin^- + score^-) + activation(margin^+ - score^+)
+
+    ---
+    name: Double Margin
+    """
+
+    hpo_default: ClassVar[Mapping[str, Any]] = dict(
+        margin_positive=dict(type=float, low=-1, high=1),
+        margin_negative=dict(type=float, low=-1, high=1),
+        positive_negative_balance=dict(type=float, low=1.0e-03, high=1.0 - 1.0e-03),
+        margin_activation=dict(
+            type='categorical',
+            choices=margin_activation_resolver.options,
+        ),
+    )
+
+    def __init__(
+        self,
+        positive_margin: float = 1.0,
+        negative_margin: float = 0.0,
+        positive_negative_balance: float = 0.5,
+        margin_activation: Hint[nn.Module] = 'relu',
+        reduction: str = 'mean',
+    ):
+        r"""Initialize the double margin loss.
+
+        :param positive_margin:
+            The (absolute) margin for the positive scores. Should be larger than the negative one.
+        :param negative_margin:
+            The (absolute) margin for the negative scores. Should be smaller than the positive one.
+        :param positive_negative_balance:
+            The balance between positive and negative term. Must be in (0, 1).
+        :param margin_activation:
+            A margin activation. Defaults to ``'relu'``, i.e. $h(\Delta) = max(0, \Delta + \lambda)$, which is the
+            default "margin loss". Using ``'softplus'`` leads to a "soft-margin" formulation as discussed in
+            https://arxiv.org/abs/1703.07737.
+        :param reduction:
+            The name of the reduction operation to aggregate the individual loss values from a batch to a scalar loss
+            value. From {'mean', 'sum'}.
+        """
+        super().__init__(reduction=reduction)
+        if positive_negative_balance <= 0 or positive_negative_balance >= 1:
+            raise ValueError(f"The positive-negative balance weight must be in (0, 1), but is {positive_negative_balance}")
+        if positive_margin < negative_margin:
+            logger.warning("Positive margin should not be smaller than negative margin. Swapping them.")
+            positive_margin, negative_margin = negative_margin, positive_margin
```

Comment:
Maybe we should fail here. If we do, designing the search space is a bit tedious since we need to consider this dependency.

We could also choose an alternative parameterization, e.g.
```python
positive_margin = negative_margin + offset
```
or
```python
negative_margin = middle - offset
positive_margin = middle - offset
```

each with `offset >= 0`.

Reply:
done in 6a5c1c3

Reply:
Should I consider these comments?

Reply:
If you had liked the other option better, yes 😉

Reply:
I think it's best we fail on bad input rather than guess what the issues are. I like having them in absolute rather than relative also

Reply:
I still kept the variation of one of the margins & offset, mainly because this makes defining search spaces more convenient, but the resolution method fails if there is any ambiguity.